### DISCUSSION TITLE
"The depot will be dropped when all process for this request done" --> What "dropped" means here?

### DISCUSSION THREAD

**Discussion by sidoqi** [salvo-rs/salvo#675] [Q&A]
I read from Depot documentation: https://salvo.rs/book/core/depot.html

**Comment by sidoqi** [comment]
Additionally, can we conclude that Depot is equivalent to AppState in Axum?

**Comment by chrislearn** [comment]
Means that the instance of depot is destroyed and the memory will be released.

I don't know much about axum.

**Comment by sidoqi** [comment]
Thanks @chrislearn, What I really want to make sure is whether `Depot` is meant to store some app state (eg: DB pool configuration) and you can propagate it between handlers/middlewares?

If not, how we perform dependency injection in Salvo? Thanks

**Reply by chrislearn** [reply]
Use Affix, see this example: https://github.com/salvo-rs/salvo/blob/main/examples/affix/src/main.rs

**Reply by sidoqi** [reply]
Thanks, so using affix will prevent depot to drop the dependencies after the request is done, right?
